### PR TITLE
Add Travis CI for ANTLR tests

### PR DESCRIPTION
**Pull request by zachschoenfeld33**
This PR adds CI support to the OpenQASM repo by verifying that new PR's pass the ANTLR tests. This, of course, requires that any updates to the specification are reflected in the example files.

### CODE REVIEW

File: source/grammar/README.md [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
+3. Generate the ANTLR parser files in Python: `antlr4 -Dlanguage=Python3 MYPATH/qasm3.g4`
+    - Note: This assumes you set the `antlr4` alias on installation.
+4. You can now use the generated files to parse qasm3 code! See, for instance, the method `build_parse_tree()` in `openqasm/source/grammar/tests/test_grammar`.
 
-Run Tests
-    - Run the test suite: `python -m unittest MYPATH/test_grammar.py`. The `test_grammar.py` file is
-    located at: `openqasm/source/grammar/tests/test_grammar.py`.
+## Run the Tests
+1. Make sure you are set up with ANTLR by following the steps above.
+2. Run the test suite: `pytest MYPATH/test_grammar.py` (or just `pytest`)
```
Comment (taalexander) [inline source/grammar/README.md:15]:
```suggestion
2. From the root of the repository run the test suite: `pytest openqasm/source/grammar/tests/test_grammar.py` (or just `pytest .`)
```

File: source/grammar/README.md [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
+    - `test_grammar.py` path: `openqasm/source/grammar/tests/test_grammar.py`.
```
Comment (taalexander) [inline source/grammar/README.md:16]:
```suggestion
```

**Review by taalexander — CHANGES_REQUESTED** [review]
Just a minor change suggested. Otherwise looks good.

File: source/grammar/README.md (hunk 2 above)
Comment (zachschoenfeld33) [reply source/grammar/README.md]: Addressed in https://github.com/Qiskit/openqasm/pull/171/commits/bc4a93aca446153a6a2484eca097946b4d5238cc.

File: source/grammar/README.md (hunk 1 above)
Comment (zachschoenfeld33) [reply source/grammar/README.md]: Addressed in https://github.com/Qiskit/openqasm/pull/171/commits/bc4a93aca446153a6a2484eca097946b4d5238cc.

**Review by taalexander — APPROVED** [review]

**Pull request merged**